openpgp/packet/private_key.go

```diff
@@ -174,27 +229,27 @@ func (pk *PrivateKey) Serialize(w io.Writer) (err error) {
 			'G', 'N', 'U', // "GNU" as a string
 			1, // Extension type 1001 (minus 1000)
 		})
+	} else if pk.Encrypted {
+		_, err = buf.Write([]byte{
+			254,             // SHA-1 Convention
+			byte(pk.cipher), // Encryption scheme
+		})
+		if err != nil {
```

Comment:
I'm wondering why are we checking for errors here, when `buf` is `bytes.Buffer`. I even checked https://golang.org/src/bytes/buffer.go and it does not seem that Write operations on Buffer ever end up with an error. @maxtaco am I missing something? Or is it a good convention to always check for errors?